.github/workflows/build.yml

```diff
@@ -0,0 +1,89 @@
+name: Release
+
+on: push
+
+jobs:
+  binary:
+    if: startsWith(github.ref, 'refs/tags/v')
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.24.x
+          stable: true
+
+      - name: Get dependencies
+        run: go get -v -t -d ./...
+
+      - name: Build
+        run: |
+          TAG=$(git describe --tags --abbrev=0)
+          GIT_TAG=$TAG make clean build
+
+      - name: Create release
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: build/*
+          tag: ${{ github.ref }}
+          overwrite: true
+          file_glob: true
+
+  image:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Store version in environment variable if this is a release tag
+        if: startsWith(github.ref, 'refs/tags/v')
+        run: echo "VERSION=${GITHUB_REF/refs\/tags\/v/}" >> $GITHUB_ENV
+
+      - name: Enable images push
+        if: ${{ startsWith(github.ref, 'refs/heads/master') || startsWith(github.ref, 'refs/tags/v') }}
+        run: echo "DOCKER_BUILD_PUSH=true" >> $GITHUB_ENV
+
+      - name: Store commit in environment variable
+        run: echo "COMMIT=${{ github.sha }}" >> $GITHUB_ENV
+
+      - name: Setup build flags
+        run: |
+          export LD_FLAGS="-w -s -X main.Version=$VERSION -X main.BuildDate=$(date "+%F-%T") -X main.Commit=$(git rev-parse --verify HEAD) -X main.Mode=prod"
```

Comment:
For `make build` this is done directly in the makefile, for the docker build step this should be done similarly.

Reply:
I tried to stick to what was done in the server project, assuming it was the reference project of gotify.  Do I understand correctly that this was not a good idea and this repo follows its own evolution path?  I have updated it now to set it in the makefile